src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/SqlServerType.cs

```diff
@@ -32,11 +32,23 @@ public class ServerVersionHelper
         /// <summary>
         /// Converts a server type to ValidForFlag
         /// </summary>
-        public static ValidForFlag GetValidForFlag(SqlServerType serverType)
+        public static ValidForFlag GetValidForFlag(SqlServerType serverType, Database database = null)
+        {
+            return GetValidForFlag(serverType, database != null && database.IsSqlDw);
+        }
+
+        /// <summary>
+        /// Converts a server type to ValidForFlag
+        /// </summary>
+        public static ValidForFlag GetValidForFlag(SqlServerType serverType, bool isSqlDw)
         {
             ValidForFlag validforFlag = ValidForFlag.All;
             if (Enum.TryParse<ValidForFlag>(serverType.ToString(), out validforFlag))
             {
+                if (isSqlDw)
+                {
+                    validforFlag = validforFlag | ValidForFlag.SqlDw;
```

Comment:
I think here it needs to be a simple `validForFlag = ValidForFlag.SqlDw`. The server/DB should have exactly 1 value (e.g. 2016, 2017, Azure SQL DB) so it can be checked, whereas the property filters can have many (e.g. they can be SQL2016 | SQL2017 | AzureSqlDb).